### PR TITLE
rucio: api and unit test

### PR DESCRIPTION
**Pull request by TomasJavurek**
next try

### CODE REVIEW

File: pilot/test/test_copytools_rucio.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+from pilot.util.tracereport import TraceReport
+
+
+def check_env():
+    """
+    Function to check whether rucio copytool is loaded correctly.
+    To be used to decide whether to skip some test functions.
+    :returns True: if rucio copytool is available. Otherwise False.
+    """
+    aval = False
```
Comment (anisyonk) [inline pilot/test/test_copytools_rucio.py]: Tomas, here you can simply try to import Rucio client to decide if rucio environment is available (in order to run tests at proper machine).

Comment (TomasJavurek) [reply pilot/test/test_copytools_rucio.py]: Ok. Cool. There is the problem with the logs at the moment, that has to be solved. Namely, that this version of rucio copytool produces empty logs.

File: pilot/copytool/rucio_api.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
+#!/usr/bin/env python
```
Comment (anisyonk) [inline pilot/copytool/rucio_api.py]: Tomas, why did you clone the copytool?

Comment (TomasJavurek) [reply pilot/copytool/rucio_api.py]: Ah, sorry, that was just my local testing version.

**Comment by PalNilsson** [comment]
Hi, please check why these jobs failed:
https://bigpanda.cern.ch/jobs/?computingsite=CERN-PROD_UCORE_2&hours=6&display_limit=100&timestamp=14:10:54
The successful one was run by the current prod version so there is no problem reading the file that the other jobs had issues with.

**Comment by TomasJavurek** [comment]
Hi Paul, 

job id, or prodsourcelable pls.

**Comment by PalNilsson** [comment]
Hi, the job ids are in the link above, but now you will have to update the time since it only shows the last 6 hours. Here's the link with the jobs the last 24 hours:
https://bigpanda.cern.ch/jobs/?computingsite=CERN-PROD_UCORE_2&hours=24&display_limit=100
I think we had the problem before but I forgot exactly what the solution was..

**Comment by PalNilsson** [comment]
BTW we (me and Wen) are working on two urgent patches so we will probably wait again with merging this. Anyway I can test it separately like with the jobs above.

**Comment by TomasJavurek** [comment]
The logs seems to be empty. But the error message is pointing to transformation, right?

**Comment by TomasJavurek** [comment]
I would ask you to try asap. I need to introduce pcaches and test storm sites with that, plus, there is request of passing info from Rucio at file level, that I need to implement.

**Comment by PalNilsson** [comment]
Comparing the successful job with one of the failed ones;
https://bigpanda.cern.ch//media/filebrowser/8938e7af-8ec2-44ad-bb1a-941bbe9344f0/panda/tarball_PandaJob_4355163789_CERN-PROD_UCORE_2/pilotlog.txt
vs
https://bigpanda.cern.ch//media/filebrowser/8863c080-1b67-4ffa-a51b-01f9ff8977bb/panda/tarball_PandaJob_4355731624_CERN-PROD_UCORE_2/pilotlog.txt

I see this:
2019-05-21 14:12:16,464 | WARNING  | copytool_in         | pilot.copytool.rucio             | copy_in                   | file does not exist: /pool/condor/dir_12693/tmp/atlas_d1S8WgVr/PanDA_Pilot-4355731624/HITS.06828093._000096.pool.root.1 (cannot verify catalog checksum)

I think the local file path is not correct:
2019-05-21 14:12:16,451 | DEBUG    | copytool_in         | pilot.copytool.rucio             | _download_item            | renaming '/pool/condor/dir_12693/tmp/atlas_d1S8WgVr/HITS.06828093._000096.pool.root.1.part' to '/pool/condor/dir_12693/tmp/atlas_d1S8WgVr/HITS.06828093._000096.pool.root.1'

but the code is looking in:
/pool/condor/dir_12693/tmp/atlas_d1S8WgVr/PanDA_Pilot-4355731624/HITS.06828093._000096.pool.root.1

Please check.

**Comment by TomasJavurek** [comment]
Checking that.

But that shouldn't be a problem. When downloading, rucio is placing the file into filename.part, and it is renamed to just filename at the end of download. Thus there is the logger hit about the renaming. Checksum in pilot is checked only after this process finishes (all threads from downloadclient).

**Comment by TomasJavurek** [comment]
ah, I see the subfolder /PanDA_Pilot-4355731624 is missing.

**Comment by TomasJavurek** [comment]
right Paul, thanks for finding this. There is something fishy directly in the copytool, because:

2019-05-21 14:12:10,316 | INFO     | copytool_in         | pilot.copytool.rucio             | copy_in                   | the file will be stored in /pool/condor/dir_12693/tmp/atlas_d1S8WgVr/PanDA_Pilot-4355731624

but later:

2019-05-21 14:12:10,553 | INFO     | copytool_in         | pilot.copytool.rucio             | _stage_in_api             | _stage_in_api file: {'did_scope': 'mc15_13TeV', 'did': 'mc15_13TeV:HITS.06828093._000096.pool.root.1', 'rse': u'CERN-PROD_DATADISK', 'no_subdir': True, 'pfn': u'root://eosatlas.cern.ch:1094//eos/atlas/atlasdatadisk/rucio/mc15_13TeV/6a/54/HITS.06828093._000096.pool.root.1', 'did_name': 'HITS.06828093._000096.pool.root.1', 'transfer_timeout': 489, 'base_dir': '/pool/condor/dir_12693/tmp/atlas_d1S8WgVr'}

where base_dir is already missing the /PanDA_Pilot-4355731624

**Comment by TomasJavurek** [comment]
yep:

Python 2.7.10 (default, Feb  7 2017, 00:08:15) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from os.path import dirname
>>> dst = '/pool/condor/dir_12693/tmp/atlas_d1S8WgVr/PanDA_Pilot-4355731624'
>>> print dirname(dst)
/pool/condor/dir_12693/tmp/atlas_d1S8WgVr

**Comment by TomasJavurek** [comment]
It wasn't really a bug, bacause we have that in pilot1 as well:
https://github.com/TomasJavurek/pilot/blob/main-dev/movers/rucio_sitemover.py#L143

we are just not doing the checksum there and we work directly with dst where it was downloaded.

**Comment by PalNilsson** [comment]
ok it works now,
https://bigpanda.cern.ch/job?pandaid=4357185164
will start HC testing tomorrow. it will take some time because i need to create a new template with current pilot2 sites. we need that anyway.

**Comment by TomasJavurek** [comment]
Thx a lot Paul. Once we are sure this works, I'll start to implement things for pcaches.

**Comment by mlassnig** [comment]
@PalNilsson hi paul, can you please merge this, we need to get rid of the old command line tools. thx!

**Comment by PalNilsson** [comment]
Hi yes we can merge it now

**Pull request merged**